src/main.cpp

```diff
@@ -149,7 +149,7 @@ int main(int argc, char *argv[]) {
          ndesign_global);
 
   /* Initialize Hessian approximation */
-  HessianApprox *hessian = 0;
+  HessianApprox *hessian;
```

Comment:
I guess it'd be a good idea to set this pointer to NULL? Otherwise it might be unassigned after the switch...

Reply:
That's why I added the `default` case to the switch clause. Just read on. ;)

Here is what you have to look for:
```
    default:
      printf("Error: unexpected hessianapprox_type returned");
      return 0;
  }
```
Further `0 == NULL` evaluates to false. `NULL` (non legal address) is not `0` (an integer value). Sometimes `0` is overridden by libraries to be a *null pointer constant*, but I prefer the solution above as it is more clear and does not have any hidden assumptions.

Reply:
Otherwise your point is completely valid. :)

Reply:
Otherwise your point(er) is completely valid. :)

Reply:
OK, sorry didn't see that. Let's still set it to NULL then. Shoudn't hurt...